authentik/brands/api.py

```diff
@@ -0,0 +1,142 @@
+"""Serializer for brands models"""
+from typing import Any
+
+from django.db import models
+from drf_spectacular.utils import extend_schema
+from rest_framework.decorators import action
+from rest_framework.exceptions import ValidationError
+from rest_framework.fields import CharField, ChoiceField, ListField
+from rest_framework.filters import OrderingFilter, SearchFilter
+from rest_framework.permissions import AllowAny
+from rest_framework.request import Request
+from rest_framework.response import Response
+from rest_framework.serializers import ModelSerializer
+from rest_framework.viewsets import ModelViewSet
+
+from authentik.api.authorization import SecretKeyFilter
+from authentik.brands.models import Brand
+from authentik.core.api.used_by import UsedByMixin
+from authentik.core.api.utils import PassiveSerializer
+from authentik.tenants.utils import get_current_tenant
+
+
+class FooterLinkSerializer(PassiveSerializer):
+    """Links returned in Config API"""
+
+    href = CharField(read_only=True)
+    name = CharField(read_only=True)
+
+
+class BrandSerializer(ModelSerializer):
+    """Brand Serializer"""
+
+    def validate(self, attrs: dict[str, Any]) -> dict[str, Any]:
+        if attrs.get("default", False):
+            brands = Brand.objects.filter(default=True)
+            if self.instance:
+                brands = brands.exclude(pk=self.instance.pk)
+            if brands.exists():
+                raise ValidationError({"default": "Only a single brand can be set as default."})
+        return super().validate(attrs)
+
+    class Meta:
+        model = Brand
+        fields = [
+            "brand_uuid",
+            "domain",
+            "default",
+            "branding_title",
+            "branding_logo",
+            "branding_favicon",
+            "flow_authentication",
+            "flow_invalidation",
+            "flow_recovery",
+            "flow_unenrollment",
+            "flow_user_settings",
+            "flow_device_code",
+            "event_retention",
```

Comment:
This is probably one of the configuration options we can move to the tenant instead of the brand, it was only added here to not have it be a YAML-based config option

Reply:
How would we migrate the current config to tenants? I'm thinking just drop it, and start fresh, adding it to release notes as a breaking change.

Reply:
I think so too, yeah

Reply:
Done in https://github.com/goauthentik/authentik/pull/7590/commits/66ba607317a4722b6af936fc9275738f0912357c